plugins/catalog-backend/src/service/DefaultEntitiesCatalog.ts

```diff
@@ -246,21 +248,49 @@ export class DefaultEntitiesCatalog implements EntitiesCatalog {
     // means it'll never try to write down the children again (it assumes that
     // they already exist). This means that without the code below, the database
     // never "heals" from accidental deletes.
-    await this.database<DbRefreshStateRow>('refresh_state')
-      .update({
-        result_hash: 'child-was-deleted',
-        next_update_at: this.database.fn.now(),
-      })
-      .whereIn('entity_ref', function parents(builder) {
-        return builder
-          .from<DbRefreshStateRow>('refresh_state')
-          .innerJoin<DbRefreshStateReferencesRow>('refresh_state_references', {
-            'refresh_state_references.target_entity_ref':
-              'refresh_state.entity_ref',
-          })
-          .where('refresh_state.entity_id', '=', uid)
-          .select('refresh_state_references.source_entity_ref');
-      });
+    if (dbConfig.client.includes('mysql')) {
```

Comment:
i tried rewriting this as a single query but it just got worse - ugh, this is a really silly limitation in mysql :/

Reply:
Yeah, I know. I spent a fair amount of time trying to work around it too, to no avail.